File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {
    fun convert(input: Int): String = buildString {
        if (input hasFactor 3) append("Pling")
        if (input hasFactor 5) append("Plang")
        if (input hasFactor 7) append("Plong")
        if (isEmpty()) input.toString()
    }

    private infix fun Int.hasFactor(factor: Int): Boolean = this % factor == 0
}

File: dart/raindrops/lib/raindrops.dart
class Raindrops {
  String convert(int input) {
    var raindrops = "";
    if (input % 3 == 0) raindrops += "Pling";
    if (input % 5 == 0) raindrops += "Plang";
    if (input % 7 == 0) raindrops += "Plong";
    if (raindrops.isEmpty) raindrops = "$input";
    return raindrops;
  }
}
